### DISCUSSION TITLE
Conditional Step

### DISCUSSION THREAD

**Discussion by samuelbenison** [getgauge/gauge#2101] [General]
Is it possible to create a conditional step and run a few sub-steps only if the conditional step is passed?

Currently, I have to fail the step so the steps below won't execute. I am trying to see if there is a way to skip the rest of the steps from the first step without failing the scenario?

Is it possible? Curious to see if there is a way to do it.

**Comment by sriv** [comment]
There are two aspects to your question (and neither is possible with gauge at the moment, but I just wanted to call this out)

1. ability to abort the execution flow without failing the scenario
2. conditional execution capability - this is different IMO, since this can lead to branching of steps (i.e. if-then-else capability)

On (2):
Gauge does not try to bring in logic/programming into the markdown (except for some parameter stuff). The idea is to keep it such that scenarios can be read and understood without any runtime context. Bringing in any dependency on conditional statements/iterators etc will compromise this.

On (1):
This can be introduced but it needs to be thought through in terms of reporting. What would an aborted scenario signify? Should the scenario be considered as passed, failed or do we need a new state?

**Reply by samuelbenison** [reply]
Yes, that makes sense, point # 2 is not really required.

But regarding # 1, it would be definitely useful to have a neutral third state like **Skipped** or **Blocked**, and use that state for step aborting.

That will open up a lot of doors in our testing. We can use the first step to decide whether to execute or skip that scenario.

Is that a simple feature to add? I can create a feature ticket if you would like.

**Comment by sriv** [comment]
> Is that a simple feature to add? I can create a feature ticket if you would like.

It is not very straightforward - this needs change in gauge, all language runners as well as reporting.
